nixos/modules/programs/light.nix

```diff
@@ -17,11 +18,82 @@ in
           and udev rules granting access to members of the "video" group.
         '';
       };
+
+      minimumBrightness = mkOption {
+        type = types.int;
+        default = 1;
+        description = ''
+          Set minimum cap on the brightness value, as some controllers set
+          the display to be pitch black at a 0 (or higher).
+        '';
+      };
+
+      devicePath = mkOption {
+        type = with types; nullOr string;
+        default = null;
+        example = "sysfs/backlight/intel_backlight";
+        description = ''
+          Specify device target path.  Use <command>light -L</command> to
+          list available devices.
+        '';
+      };
+
+      mediaKeys = {
+
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to enable brightness control with keyboard media keys.
+            You want to leave this disabled if you run a desktop environment
+            like KDE, Gnome, Xfce, etc, as those handle such things themselves.
+            You might want to enable this if you run a minimalistic desktop
+            environment or work from bare linux ttys/framebuffers.
+            Enabling this will turn on <option>services.actkbd</option>.
+          '';
+        };
+
+        step = mkOption {
+          type = types.int;
+          default = 1;
+          description = ''
+            The value by which to increment/decrement brightness on media keys.
+            See light(1) for allowed values.
+          '';
+        };
+
+      };
+
     };
   };
 
   config = mkIf cfg.enable {
     environment.systemPackages = [ pkgs.light ];
     services.udev.packages = [ pkgs.light ];
+
+    services.actkbd = mkIf cfg.mediaKeys.enable {
+      enable = true;
+      bindings = [
+        { keys = [ 224 ]; events = [ "key" "rep" ]; command = "${light} -U ${toString cfg.mediaKeys.step}"; }
```

Comment:
I'm not sure we actually want to include these sorts of options in NixOS as they increase evaluation time for each user, and are not very complicated. I would prefer we add documentation to the wiki on how to do this.

Reply:
Or maybe this is better fit for [NUR](https://github.com/nix-community/NUR)

Reply:
@grahamc What's the purpose of light module then? Also there is an inconsistency for laptop users. We have `sound.mediaKeys` but have no brightness keys control.